src/Library.php

```diff
@@ -183,4 +199,54 @@ public static function delete_expired_draft_orders() {
 			"
 		);
 	}
+
+	/**
+	 * Attempt to process a payment for the checkout API if no payment methods support the
+	 * woocommerce_rest_checkout_process_payment_with_context action.
+	 *
+	 * @param PaymentContext $context Holds context for the payment.
+	 * @param PaymentResult  $result  Result of the payment.
+	 */
+	public static function process_legacy_payment( PaymentContext $context, PaymentResult &$result ) {
+		if ( $result->status ) {
+			return;
+		}
+
+		// phpcs:ignore WordPress.Security.NonceVerification
+		$post_data = $_POST;
+
+		// Set constants.
+		wc_maybe_define_constant( 'WOOCOMMERCE_CHECKOUT', true );
+
+		// Add the payment data from the API to the POST global.
+		$_POST = $context->payment_data;
+
+		// Call the process payment method of the chosen gatway.
+		$available_gateways = WC()->payment_gateways->get_available_payment_gateways();
+
+		if ( ! isset( $available_gateways[ $context->payment_method ] ) ) {
+			return;
+		}
+
+		$payment_method_object = $available_gateways[ $context->payment_method ];
+		$payment_method_object->validate_fields();
+
+		if ( 0 !== wc_notice_count( 'error' ) ) {
+			return;
+		}
```

Comment:
This is something we probably will have to handle somehow right? If there is a notice, then there was an error, so we'll need some way of communicating that error back to the client?

Reply:
Yes. I'm not sure how yet, I just added it here "just in case". I'd consider this BW compat code experimental.